Explorer/Assets/Scripts/CrdtEcsBridge/JsModulesImplementation/ClientWebSocketApiImplementation.cs

```diff
@@ -0,0 +1,225 @@
+        private readonly IJsOperations jsOperations;
+
+        private readonly ConcurrentDictionary<int, WebSocketRental> webSockets = new ();
+
+        private int nextId;
+
+        public ClientWebSocketApiImplementation(IInstancePoolsProvider instancePoolsProvider, IJsOperations jsOperations)
+        {
+            this.instancePoolsProvider = instancePoolsProvider;
+            this.jsOperations = jsOperations;
+        }
+
+        public void Dispose()
+        {
+            foreach (WebSocketRental rental in webSockets.Values)
+                rental.Dispose();
+
+            webSockets.Clear();
+        }
+
+        public int CreateWebSocket(string url)
+        {
+            Interlocked.Increment(ref nextId);
+
+            webSockets[nextId] = new WebSocketRental(); // ClientWebSocket does not support reviving
+            return nextId;
+        }
+
+        public async UniTask ConnectAsync(int websocketId, string url, CancellationToken ct)
+        {
+            await GetInstanceOrThrow(websocketId).WebSocket.ConnectAsync(new Uri(url), ct);
+        }
+
+        public async UniTask SendBinaryAsync(int websocketId, IArrayBuffer data, CancellationToken ct)
+        {
+            WebSocketRental webSocket = GetInstanceOrThrow(websocketId);
+
+            var bytesCount = (int)data.Size;
+            if (bytesCount == 0) return;
+
+            using PoolableByteArray poolableArray = instancePoolsProvider.GetCrdtRawDataPool(bytesCount);
+
+            data.ReadBytes(0, data.Size, poolableArray.Array, 0);
+            await CHUNK_TRANSMISSION.SendAsync(webSocket, poolableArray.Memory, WebSocketMessageType.Binary, ct);
+        }
+
+        public async UniTask SendTextAsync(int websocketId, string data, CancellationToken ct)
+        {
+            WebSocketRental webSocket = GetInstanceOrThrow(websocketId);
+
+            int utfBytesCount = Encoding.UTF8.GetByteCount(data);
+
+            if (utfBytesCount == 0) return;
+
+            using PoolableByteArray poolableArray = instancePoolsProvider.GetCrdtRawDataPool(utfBytesCount);
+
+            Encoding.UTF8.GetBytes(data, poolableArray.Memory.Span);
+            await CHUNK_TRANSMISSION.SendAsync(webSocket, poolableArray.Memory, WebSocketMessageType.Text, ct);
+        }
+
+        public async UniTask CloseAsync(int websocketId, CancellationToken ct)
+        {
+            if (!webSockets.TryGetValue(websocketId, out WebSocketRental webSocket))
+                throw new ArgumentException($"WebSocket with id {websocketId} does not exist.");
+
+            await webSocket.WebSocket.CloseAsync(WebSocketCloseStatus.NormalClosure, string.Empty, ct);
+        }
+
+        public async UniTask<IWebSocketApi.ReceiveResponse> ReceiveAsync(int websocketId, CancellationToken ct)
+        {
+            WebSocketRental webSocket = GetInstanceOrThrow(websocketId);
+
+            (PoolableByteArray result, WebSocketMessageType messageType, WebSocketCloseStatus closeStatus)
```

Comment:
Can receive throw and then we have a floating instance of WebSocket from the pool?

Reply:
We don't pool WebSockets - according to the .NET API they are not reusable